flink-runtime/src/main/java/org/apache/flink/runtime/operators/util/BitSet.java

```diff
@@ -89,8 +88,13 @@ public int bitSize() {
 	 * Clear the bit set.
 	 */
 	public void clear() {
-		for (int i = 0; i < byteLength; i++) {
-			memorySegment.put(offset + i, (byte) 0);
+		int index = 0;
+		while (index < byteLength) {
```

Comment:
This looks like a potential bug to me, assuming that we can always still add 8 bytes if we are smaller than `byteLength` can go out of bounds IMO. Or in case that it could not go out of bounds because it is always a multiple of 8, the second loop that follows does not make sense and it is currently also never invoked. I would then suggest to add a test that covers clearing from a bitset with an amount of bits that is not divisible by 8.

Reply:
@StefanRRichter , thanks a lot for your careful review. 

It is a real bug. I have fixed it. In addition, I have added test cases by making BitSetTest parameterized. So the clear method is tested against different amounts of bits.